enigma-principal/app/src/common_u/custom_errors.rs

```diff
@@ -0,0 +1,104 @@
+use std::sync::PoisonError;
+
+use thiserror::Error;
+use failure::Error as GenericErr;
+use sgx_types::sgx_status_t;
+
+use enigma_types;
+use enigma_tools_u::common_u::errors::Web3Error;
+use enigma_tools_m::ToolsError;
+#[derive(Error, Debug)]
+pub enum ControllerError {
+    #[error(transparent)]
+    ConfigError(#[from] ConfigError),
+    #[error(transparent)]
+    HTTPServerError(#[from] HTTPServerError),
+    #[error("Error in Controller: {0}")]
+    VerifierError(VerifierError),
+    #[error("Error in Controller: {0}")]
+    EnclaveError(EnclaveError),
+    #[error("Error in Controller: {0}")]
+    EpochError(EpochError),
+    #[error(transparent)]
+    ContractError(#[from] Web3Error),
+    #[error("Error while trying to produce quote")]
+    QuoteErr,
+    #[error("the error received is: {0}")]
+    GenericError(GenericErr)
+}
+
+#[derive(Error, Debug)]
+pub enum ConfigError {
+    #[error("Unable to open config file")]
+    FileDoesntExist,
+    #[error("Unable to convert file content to string")]
+    NotAString,
+    #[error("Cannot parse data")]
+    Parsing,
+}
+
+#[derive(Error, Debug)]
+pub enum HTTPServerError {
+    #[error("input signature got an unexpected size (65 bytes): {0} bytes")]
```

Comment:
```suggestion
    #[error("Input signature got an unexpected size: {0} bytes. Expected 65 bytes.")]
```
Generally we should probably capitalize the Display messages for good style, but it's not very important, all things considered.